include/misc/statistics.hpp

```diff
@@ -0,0 +1,53 @@
+#pragma once
+
+#include <forward_list>
+
+#include <seqan3/io/sam_file/input.hpp>
+
+/*! \brief Returns a list of pairs, where the first integer is a random chromosome and the second is a random position.
+ *        The list is sorted by default.
+ *  \param sample_size - The number of positions to generate.
+ *  \param header - The header from the SAM file input.
+ * 
+ *  \return Returns a sorted forward_list of pairs of reference chromosome IDs and positions.
+ */
+std::forward_list<std::pair<int32_t, int32_t>> get_random_positions(uint64_t sample_size, seqan3::sam_file_header<std::deque<std::string>> & header)
+{
+    std::forward_list<std::pair<int32_t, int32_t>> output{};
+    if (sample_size == 0)
+    {
+        output.push_front(std::pair<int32_t, int32_t>{-1, -1});
+    }
+    int32_t rand_chr{};
+    int32_t rand_pos{};
+
+    for (int i = 0; i < sample_size; ++i)
+    {
+        rand_chr = std::rand() % header.ref_ids().size();
+        rand_pos = std::rand() % std::get<0>(header.ref_id_info[rand_chr]);
+
+        output.push_front(std::pair<int32_t, int32_t>{rand_chr, rand_pos});
+    }
+
+    output.sort();
+    return output;
+}
+
+/*! \brief Get the read length of a read from its cigar string.
+ *  \param cigar - The cigar according to the seqan3 representation (a vector of cigar alphabet).
+ *
+ *  \return Returns the length of the cigar string.
+ */
+
+uint32_t get_read_length(std::vector<seqan3::cigar> & cigar)
+{
+    using seqan3::get;
+
+    uint32_t result{0};
+    for (auto e : cigar)
+    {
+        result += get<0>(e);
```

Comment:
Is this correct? 

The sam specification says:

| Op | BAM | Description | Consumes query | Consumes reference |
| -- | --- | ----------- | -------------- | ------------------ |
| M  | 0   | alignment match (can be a sequence match or mismatch) | yes | yes |
| I  | 1   | insertion to the reference |yes | no |
| D  | 2   | deletion from the reference | no | yes |
| N  | 3   | skipped region from the reference | no | yes |
| S  | 4   | soft clipping (clipped sequences present in SEQ) | yes | no |
| H  | 5   | hard clipping (clipped sequences NOT present in SEQ) | no | no |
| P  | 6   | padding (silent deletion from padded reference) | no | no |
| =  | 7   | sequence match | yes | yes |
| X  | 8   | sequence mismatch | yes | yes |

“Consumes query” and “consumes reference” indicate whether the CIGAR operation causes the alignment to step along the query sequence and the reference sequence respectively.

So I would say that you over-count the read length.

https://samtools.github.io/hts-specs/SAMv1.pdf#page=8